### ISSUE TITLE
AttributeError ('bool' object has no attribute 'needFile')

### ISSUE THREAD

**Issue by jesseray** [HelloZeroNet/ZeroNet#415]
I just downloaded the Lantern installers (https://getlantern.org) to make them available on ZeroNet. I added them to my ZeroNet site (17EcoSLuGdJKWrKnBDN1qVsyxN2dZqtJQL), signed, and published. I checked whether everything was working, but it wasn't. Clicking on one of my ZeroNet links to download an installer (any installer), my browser shows me this error message:

> Server error
> 
> Err: AttributeError: 'bool' object has no attribute 'needFile' in UiServer.py line 81 > UiRequest.py line 82 > UiRequest.py line 190 > UiRequestPlugin.py line 22 > UiRequest.py line 352
> 
> Please report it if you think this an error.
> 
> Details:
> 
> {
>     "GATEWAY_INTERFACE": "CGI/1.1", 
>     "HTTP_ACCEPT": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8", 
>     "HTTP_ACCEPT_ENCODING": "gzip, deflate, sdch", 
>     "HTTP_ACCEPT_LANGUAGE": "en-US,en;q=0.8,fr;q=0.6", 
>     "HTTP_CONNECTION": "keep-alive", 
>     "HTTP_DNT": "1", 
>     "HTTP_HOST": "127.0.0.1:43110", 
>     "HTTP_UPGRADE_INSECURE_REQUESTS": "1", 
>     "HTTP_USER_AGENT": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/49.0.2623.108 Chrome/49.0.2623.108 Safari/537.36", 
>     "PATH_INFO": "/downloads/lantern/lantern-installer-beta.apk", 
>     "QUERY_STRING": "", 
>     "REMOTE_ADDR": "127.0.0.1", 
>     "REMOTE_PORT": "58247", 
>     "REQUEST_METHOD": "GET", 
>     "SCRIPT_NAME": "", 
>     "SERVER_NAME": "localhost", 
>     "SERVER_PORT": "43110", 
>     "SERVER_PROTOCOL": "HTTP/1.1", 
>     "SERVER_SOFTWARE": "gevent/1.0 Python/2.7", 
>     "arguments": {
>         "action": "main", 
>         "batch": false, 
>         "bit_resolver": "1Name2NXVi1RDPDgf5617UoW7xA6YrhM9F", 
>         "coffeescript_compiler": null, 
>         "config_file": "zeronet.conf", 
>         "connected_limit": 15, 
>         "data_dir": "data", 
>         "debug": false, 
>         "debug_gevent": false, 
>         "debug_socket": false, 
>         "disable_db": false, 
>         "disable_encryption": false, 
>         "disable_sslcompression": true, 
>         "disable_udp": false, 
>         "fileserver_ip": "*", 
>         "fileserver_port": 15441, 
>         "homepage": "1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D", 
>         "ip_external": null, 
>         "keep_ssl_cert": false, 
>         "log_dir": "log", 
>         "max_files_opened": 2048, 
>         "msgpack_purepython": true, 
>         "open_browser": null, 
>         "proxy": null, 
>         "size_limit": 10, 
>         "stream_downloads": false, 
>         "tor": "enable", 
>         "tor_controller": "127.0.0.1:9051", 
>         "tor_proxy": "127.0.0.1:9050", 
>         "trackers": [
>             "zero://boot3rdez4rzn36x.onion:15441", 
>             "zero://boot.zeronet.io#f36ca555bee6ba216b14d10f38c16f7769ff064e0e37d887603548cc2e64191d:15441", 
>             "udp://tracker.coppersurfer.tk:6969", 
>             "udp://tracker.leechers-paradise.org:6969", 
>             "udp://9.rarbg.com:2710", 
>             "http://tracker.aletorrenty.pl:2710/announce", 
>             "http://explodie.org:6969/announce", 
>             "http://torrent.gresille.org/announce"
>         ], 
>         "trackers_file": false, 
>         "ui_ip": "127.0.0.1", 
>         "ui_port": 43110, 
>         "ui_restrict": false, 
>         "use_openssl": true, 
>         "use_tempfiles": false, 
>         "verbose": false
>     }, 
>     "plugins": [
>         "Sidebar", 
>         "Newsfeed", 
>         "Trayicon", 
>         "Stats", 
>         "CryptMessage", 
>         "Zeroname", 
>         "AnnounceZero"
>     ], 
>     "version_gevent": "1.0", 
>     "version_python": "2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]", 
>     "version_zeronet": "0.3.7 r1210", 
>     "wsgi.url_scheme": "http"
> }

**Comment by HelloZeroNet** [comment]
For some unknown reasons zeronet does not see your site. 
Does the 17EcoSLuGdJKWrKnBDN1qVsyxN2dZqtJQL directory exists in your data directory? does it has index.html and content.json file?
Please try to restart your zeronet client, maybe it helps.

**Comment by jesseray** [comment]
I was afraid of that. The directory and files exist. No errors on signing. (PS: documentation should be added about when the site is too big and you need to ignore files. I figured it out eventually, but that was annoying.) Unfortunately, I have had trouble opening port 15441. I have tried port forwarding, turning off the router firewall, putting my computer in DMZ mode, and turning off traffic shaping, but zeronet.py still says that the port is closed when I start up ZeroNet. This has been quite irritating :P

**Comment by jesseray** [comment]
I just learned something. I made my hyperlink wrong. I wrote `<a href='/downloads/...'>` instead of `<a href='17EcoSLuGdJKWrKnBDN1qVsyxN2dZqtJQL/downloads/...'>`. Once I fixed that, I no longer got the error message, so the error message is basically saying that the link is invalid. I think it would help to get rid of this ugly error message and replace it with a more friendly 404 File Not Found page. (Sure, you do not use HTTP and thus HTTP error codes, but you know what I mean.)
